Prepare Beta 4 Release (#2488)
apollographql/apollo-ios: CHANGELOG.md

# Change Log

## v1.0.0-beta.4

This is the fourth Beta Release of Apollo iOS 1.0. The Beta version has full feature parity with the 0.x.x releases. The API is expected to be mostly stable. Some breaking changes may occur due to user feedback prior to General Availability (GA) Release. The Beta does not have complete documentation or usage guides, which will be completed prior to GA.

This first major version will include a new code generation engine, better generated models, and many syntax and performance improvements across the entire library. The primary goal of Apollo iOS 1.0 is to stabilize the API of the model layer and provide a foundation for future feature additions and evolution of the library.

* **Breaking: Generated Files now have the file extension `.graphql.swift`.**
  * This allows you to clearly distinguish which files in your project are Apollo generated files.
  * Generated template files that are user-editable will still have the `.swift` file extension. 

>     * `CustomScalar` templates as while as the `SchemaConfiguration` file are user-editable. Once these are generated, they are never overwritten by future code generation execution.

Reviewer: ```suggestion
    * `CustomScalar` templates as well as the `SchemaConfiguration` file are user-editable. Once these are generated, they are never overwritten by future code generation execution.
```

>   * This change is also necessary for the identification of generated files for the automatic deletion functionality.

Reviewer: ```suggestion
  * This change is also necessary for the identification of generated files for the pruning functionality.
```
I recommend we stick to the same nomenclature as the configuration file.

> * **New: Automatic Deletion of Unused Generated Files**

Reviewer: ```suggestion
* **New: Pruning of Unused Generated Files**
```
